### DISCUSSION TITLE
Sorting Folders by Dates in Titles

### DISCUSSION THREAD

**Discussion by JSG-1** [SebastianMC/obsidian-custom-sort#183] [Q&A] [answered]
Hi @SebastianMC,

Thank you for this incredible plugin! I’m having trouble configuring it to handle folders with dates in their titles and would appreciate your guidance. I also use the **Folder Notes** plugin, but I know your plugin supports it, so I suspect the issue is with my configuration.

---

### Problem:

I have folders with dates in the format `YYYY-MM-DD`, which can appear **anywhere** in the folder name (start, middle, or end). I need to:

1. **Sort these folders by the date in their title**, with **newer dates at the top**.
2. If a folder does not contain a date, it should be sorted by its **creation date** (newer folders first).

---

### What I’ve Tried:

- Using `regexp: \d{4}-\d{2}-\d{2}` to match folders with dates.
- Sorting by `> created`, but this doesn’t account for the dates embedded in folder names.
- I ensured folder note metadata is present but still couldn’t extract and sort by the date in the title.

---

### Example:

**Initial Structure**:

```
2024-12-01 Important Notes  
Progress Report 2024-11-30  
Weekly Review 2024-12-03  
```

**Desired Output**:

```
Weekly Review 2024-12-03  
2024-12-01 Important Notes
Progress Report 2024-11-30  
  
```

---

Is there a way to configure the plugin to sort by dates extracted from folder names while using Folder Notes?

Thank you for your time and help!

Best regards,  
JSG

**Comment by SebastianMC** [comment]
Hi @JSG-1

you example involves a series of tricky cases:

## 1. Sorting by folder dates

In Obsidian, folders don't come with date metadata (no creation date, no modification date). By design. This prevents any sorting based on folder creation/update dates.
To overcome this limitation, the plugin can derive creation/modification from 'Folder Notes' inside a folder and treat them as folder's creation/modification date. To use this feature specify the sorting order as `advanced modified` or `advanced created`, for example `> advanced modified`.  

## 2. Use dates in titles for sorting

The plugin supports part of the scenarios which you describe.
Namely, matching against prefix of suffix numerical sequences and extracting them for sorting.
In your case the syntax would be:
- `... \-D+` to match titles ending with numerical sequences, allowing the dash `-` as a separator, and
- `\-D+ ...` to to match numerical prefix, accordingly

This works because the dates in format 'YYYY-MM-DD' can be treated as a dash-separated sequence of numbers

An example snippet of sorting specification would be then:
```yaml
  /+ \-D+ ...
  /+ ... \-D+
    > a-z
```

The additional syntax `  /+` is used to combine the two groups of files/notes and sort them as a single group.
I'm using this approach in my vaults.

At the same time, there is no support for infix matching-and-extraction of numbers. It is intentional. Overall, the plugin doesn't support free-form regexp matching against titles intentionally. It is easy to create a regexp introducing backtracking and kill the performance in result, blocking the Obsidian UI even for relatively small vaults. Matching of infix notation (e.g. dates anywhere in the title) is a good example of that.

## 3. Mixing of different sorting methods for a single-pass sorting

This is not supported, a far too advanced feature.
You can specify up to 4 sorting levels, at the same time they cannot be mixed - they are layered one on top of another.
In other words, you can specify: sort items by creation date first, then within the same date use the title sorting.

You cannot specify: for a folder take the date from tile, and if there is no date in title, take the date from metadata. Not supported.

## Recap

There are three unsupported features:
- sorting by folder creation or modification date (folders in Obsidian don't have date metadata - you can only derive the creation or modification metadata from Folder Notes)
- extracting a date from 'anywhere in the title' - only numerical date as prefix or suffix are supported
- mixing of data sources for the same sorting - only layering of up to 4 sorting methods is supported

Excluding these unsupported features, an example sorting spec could look like:

```yaml
sorting-spec: |
  target-folder: /
  /+ /folders \-D+ ...
  /+ /folder ... \-D+
    > a-z, > advanced created
```

The resulting sorting would be:
- find folders which have a dash-separated numerical sequence as a title prefix, e.g. '2024-12-01 Important Notes', AND
- folders which have a dash-separated numerical sequence as a title suffix, e.g. 'Weekly Review 2024-12-03'
- order these folders by the numerical sequence in the tile (which works correctly for dates YYYY-MM-DD)
- if two folders share the same date in title, sort them by creation date (derived from their Folder Notes)

**Comment by JSG-1** [comment]
Hi @SebastianMC 

Thank you so much for your detailed response and for taking the time to explain the plugin's capabilities and limitations. Your explanation has given me the clarity I need on how best to proceed, and I now have a much better understanding of how to configure my sorting rules effectively.

I really appreciate your help and the work you’ve put into this amazing plugin.

Best regards,
JSG

**Comment by JSG-1** [comment]
Hi @SebastianMC , 

I saw that you moved this to a discussion.  I wanted to let you know that I tried every which way to implement the change you suggested with no luck.  I also used Chat-01 to try multiple ways of getting the sort order i want with also no luck. 

Attached are my config file, a screenshot of a testing folder i created to test the sort, and a screenshot of the plugin settihngs.  Is there anything you would suggest that I could try next?  

Best regards,

JSG

[JSGsorting.md](https://github.com/user-attachments/files/18278066/JSGsorting.md)
<img width="289" alt="2025-12-30_16 06 19" src="https://github.com/user-attachments/assets/dec91cb0-439b-4bcd-abb5-c6fce375ae41" />
<img width="819" alt="image" src="https://github.com/user-attachments/assets/af160370-cd0f-4505-b3a1-55739039dc13" />

**Reply by SebastianMC** [reply]
Hi @JSG-1 

I misunderstood your answer as a success :-)
Let's continue here then

I suggest to start from scratch and apply the baby-steps approach

Here is what I did on my side, please reproduce it on your machine:

1. Create an empty vault, remove the default Welcome note
2. Install the custom sorting plugin (can be done later as well)
3. Create a folder `Testing` with three subfolders as on the screenshot below. The initial order of the folders is as visible:


<img width="568" alt="2024-12-31_10-51-46" src="https://github.com/user-attachments/assets/6c2ed10b-72ca-4ed5-b106-7e441313a6c0" />

4. In custom sorting settings enter the `JGSorting` as the name of additional sorting spec location:

<img width="1024" alt="2024-12-31_10-50-55" src="https://github.com/user-attachments/assets/ab9c6393-fd5a-431f-86a7-914422e98c97" />

5. Create the `JGSorting` file with a basic sorting specification as below, then enable the custom sorting. In result the folders should be sorted by the date suffix:
  - the sorting file itself: [JGSorting.md](https://github.com/user-attachments/files/18280676/JGSorting.md)
  - the view of the vault after applying custom sorting:
  - <img width="723" alt="2024-12-31_11-04-10" src="https://github.com/user-attachments/assets/e94b54af-9414-4efa-9b47-d91c056edfab" />

After recreating this on your side in a new blank test vault let's continue from there.
What would be the next step? Introducing folder notes?

**Comment by JSG-1** [comment]
Hi [@SebastianMC](https://github.com/SebastianMC),

Thank you for your quick response and helpful feedback!

I followed your suggestions step-by-step, and here’s what I found:

1. **Testing in a New Vault**:
    
    - I tested the suggested configuration in a new vault, and it worked as expected.
    - I then installed the **Folder Notes (FN)** plugin, set up the same configuration, and confirmed it worked in this test vault as well.
2. **Testing Nested Folders**:
    
    - I tested adding deeper folder levels, thinking this might differ from my main vault setup. After modifying `target-folder` to `/*`, the sorting still worked correctly in the test vault.
3. **Returning to My Main Vault**:
    
    - I went back to my original vault and replaced the `JSGsorting` config with just the simplified configuration you suggested, and it worked as expected.
    - However, when I re-added my full `JSGsorting` config file (which includes multiple rules for sorting various parts of my vault), the folder sorting by date stopped working.
    - Specifically, it sorts the first line (ordering INBOX, Projects, etc.) correctly, but the date-based sorting for folders fails to apply. The odd thing is that with the same config in the Test Vault, it does sort correctly. I’m not sure what could be causing the config file not to work in my main vault.

This behavior seems to suggest that the first line of the configuration might take precedence over subsequent rules.

---

### Next Steps:

What would you recommend I try next to troubleshoot this issue in my main vault? Is there a way to ensure that the folder date sorting rule is applied correctly after the initial ordering of top-level folders?

---

### Attachments:

To clarify the issue, I’ll include the following:

1. **Main Vault - Failure**: A screenshot showing the incorrect folder order after applying my full `JSGsorting` configuration.
2. **Main Vault - Simple config success** A screenshot showing how the simple configuration results in a successful sorting order.
3. **Test Vault - Success**: A screenshot of the successful sorting in the test vault using the exact same config file.
4. **Full `JSGsorting` Configuration File**: To help identify if there’s any conflict or misconfiguration.

Thank you again for your time and assistance!

Best regards,  
JSG

<img width="1103" alt="Test Vault - Success" src="https://github.com/user-attachments/assets/0ada57a9-d894-49d0-8fc7-4201c8bfbf77" />
<img width="1423" alt="Main Vault - Failure" src="https://github.com/user-attachments/assets/9a60ff6d-233c-4909-89d2-38cbac76df2b" />
<img width="1188" alt="Main Vault - Simple config success" src="https://github.com/user-attachments/assets/cc2f6f58-b66d-4022-a607-1f7330d4e1f8" />

**Comment by JSG-1** [comment]
Forgot to include the updated config file. 
[JSGsorting.md](https://github.com/user-attachments/files/18281540/JSGsorting.md)

**Reply by SebastianMC** [reply, marked as answer]
Hi @JSG-1 

thanks for the impressive amount of details and context and log of steps!
Putting all of this together, the answer lies in the complex sorting rules :-)
In this case, the relevant logic of the plugin is to always choose the more specific rules if multiple rules are specified and applicable to a folder.
In your example, the problematic folder is:
- `Projects/FUNDING/Testing`

Its subfolders aren't sorted as expected.

Then, when I read all the sorting rules in `JSGsorting` I see two applicable rules:
1. the one specified under `  target-folder: /*`
  - the `/*` means to apply a rule to all folders in the vault, regardless of the depth 
2. and the one specified under `  target-folder: Projects/*`
  - the `Project/*` tells to apply the rule for `Project` and all its descendants

Because the plugin is not capable of combining overlapping rules, it always chooses the most applicable one. In this example, the more specific rule is 2. because its scope is narrower than rule 1.  Technically, the scope of `/Projects/*` is narrower than `/*`.

The solution for this specific vault and the configuration set is to apply the rule 2. only to items in the folder `Projects`, not to any of subfolders.

Please change the scope specification:

```
  # Sorting Projects
  target-folder: Projects/*
  STRATEGY
```

to

```
  # Sorting Projects
  target-folder: Projects
  STRATEGY
```

in result, the rule 2. will no longer be applicable to subfolders of `Projects/FUNDING/Testing` and the only remaining rule will be the one which you want.

**Comment by JSG-1** [comment]
Hi [@SebastianMC](https://github.com/SebastianMC),

Thank you so much for your detailed explanation and guidance—it worked perfectly! I really appreciate your patience and the effort you put into helping me resolve this issue.

Do you happen to have a [Buy Me a Coffee](https://buymeacoffee.com/) page or a similar platform? I’d love to show my appreciation for your help and the work you’ve done on this amazing plugin.

Wishing you a very happy new year and all the best for 2025!

Best regards,
JSG

**Reply by SebastianMC** [reply]
Hi @JSG-1 

glad that it eventually worked for you!
I don't have any kind of _Buy Me a Coffee_, at the same time happy to hear your appreciation.

You too have a good new year and a fantastic entire 2025!

thanks,
Seb
